### DISCUSSION TITLE
No data for the Master and Server charts in cgi - rest are ok

### DISCUSSION THREAD

**Discussion by BillKen** [moosefs/moosefs#439] [Q&A]
![image](https://user-images.githubusercontent.com/38230457/121163080-1430a200-c881-11eb-858a-56b1d62bd0ae.png)
After some severe power glitches due to a faulty cable I have no data in the CGI Master and Server charts in either a browser or mfscli. The other charts are fine.

Ive checked for permissions and all looks ok so I am looking for suggestions on where to look next?

BillK

**Comment by chogata** [comment]
If you had a power glitch, it's entirely possible that the file(s) with chart data was not properly recorder on hard drive. No file - no charts. They should start appearing again, drawn from the moment the masters started work. If those do not show too, then there is something else wrong.
Are there files named "stats.mfs" in your /var/lib/mfs (or wherever you store local data for MooseFS processes) on your master/chunkserver machines? If they are not there, mystery solved. If they are and are bigger than 0, check permissions.

**Reply by BillKen** [reply]
There are files like this on each cunkserver:
-rw-r----- 1 mfs    mfs    4459864 Jun  8 21:00 csstats.mfs
and on the master:
-rw-r----- 1 mfs    mfs       4984552 Jun  8 21:00 stats.mfs

All stats files are current and updating, and permissions look correct.  The whole cluster has been powered off/restarted numerous times (usually powered off overnight) and the problem has existed for a few weeks as I was hoping it would fix itself.  
BillK

**Comment by stevew-Purdue** [comment]
Are you still having this problem, BillK?  What Linux distro and version are you using?

I'm experiencing the same problem on each of my MooseFS installations (I have over 10 of them).  The Master Charts and Server Charts on the CGI interface don't display any graphs.  We are using Ubuntu 20.04 and I wonder if any recent system updates have caused this failure in the display of charts in the CGI interface.

Steve

**Comment by BillKen** [comment]
Yes, still have it!
I am using current gentoo/openrc (chunkservers on a mix of stable/testing packages on arm32/arm64 and master on Intel) with moosefs from latest git across the board so its not an OS problem.  So far I have deleted all the stats files with the cluster stopped, on restart there was no change.  Permissions look good (moosefs runs as mfs:mfs).
I have run out of ideas!
BillK

**Comment by xandrus** [comment]
Hi,
I suspect that this topic is related to your python version. I mean python 3.9

Can you modify '#!' header on mfs.cgi and chart.cgi located on /usr/share/mfscgi and lock to a specific Python version?
I mean lower than 3.9? 

For example:
```
#!/usr/bin/env python3.8
 ```

**Comment by stevew-Purdue** [comment]
Hi,

That's a good suggestion but we are using Python 3.8.10 and the CGI interface displays "python: 3.8".  I did make the suggested change, though, just to be sure.  It did not make any difference.

Thanks!
Steve

________________________________

**Reply by xandrus** [reply]
Can you try python3.6?

**Comment by stevew-Purdue** [comment]
Not very easily.  We've been using Python 3.8 since we began using Ubuntu 20.04 about a year ago.  This problem just recently showed up so it might be related to a minor version update.  Looking at the logs I see an unattended upgrade occurred early yesterday of some Python packages from v. 3.8.5 to 3.8.10.

Steve

________________________________

**Comment by chogata** [comment]
Since this is a discussion thread and not an issue thread, Github is not marking this automatically, but commit 399eb8f fixes the problem for python 3.8+. @BillKen can you let us know if this works now for you too?

**Comment by stevew-Purdue** [comment]
That corrects the problem for me... thanks!

Steve


________________________________
